gateway/utils/handlers/graphql.go

```diff
@@ -9,21 +9,40 @@ import (
 
 	"github.com/gorilla/mux"
 
+	"github.com/spaceuptech/space-cloud/gateway/config"
 	"github.com/spaceuptech/space-cloud/gateway/model"
 	"github.com/spaceuptech/space-cloud/gateway/utils"
 	"github.com/spaceuptech/space-cloud/gateway/utils/graphql"
+	"github.com/spaceuptech/space-cloud/gateway/utils/syncman"
 )
 
 // HandleGraphQLRequest creates the graphql operation endpoint
-func HandleGraphQLRequest(graphql *graphql.Module) http.HandlerFunc {
+func HandleGraphQLRequest(graphql *graphql.Module, syncMan *syncman.Manager) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 
-		// Create a context of execution
-		ctx, cancel := context.WithTimeout(r.Context(), 10*time.Second)
-		defer cancel()
+		c := new(config.Project)
+		err := json.NewDecoder(r.Body).Decode(c)
+		defer utils.CloseTheCloser(r.Body)
+
+		// Throw error if request was of incorrect type
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
```

Comment:
use logrus.Errorf()
to log the errors
for reference
https://github.com/spaceuptech/space-cloud/blob/v0.16.0/runner/server/handle.go